### PR TITLE
Fix GridFS race condition

### PR DESCRIPTION
**Pull request by saghm**
The GridFS cache [get initialized with an empty chunk](https://github.com/saghm/mongo-rust-driver-prototype/blob/fix-gridfs-race-condition/src/gridfs/file.rs#L338) but the chunk itself is initialized asynchronously, which leads to a race condition when the cache is checked [here](https://github.com/saghm/mongo-rust-driver-prototype/blob/fix-gridfs-race-condition/src/gridfs/file.rs#L292) on subsequent calls to `get_chunk`. This is fixed by initializing each `CachedChunk` with a placeholder error that gets removed when the chunk is initialized.

### CODE REVIEW

**Comment by vkarpov15** [comment]
Lgtm. In case you're waiting for me to review, just feel free to merge this and future prs on your own :)

File: src/gridfs/file.rs [hunk 1]
```diff
@@ -646,7 +649,7 @@ impl CachedChunk {
         CachedChunk {
             n: n,
             data: Vec::new(),
-            err: None,
+            err: Some(Error::DefaultError(String::from("Chunk has not yet been inititalied"))),
```
Comment (kyeah) [inline src/gridfs/file.rs]: typo!!! `inititalied`

Comment (saghm) [inline src/gridfs/file.rs]: Ahhh! I'm so embarrassed

**Comment by saghm** [comment]
@kyeah, everything else look good? All the tests pass now that the other PR's have been merged

**Comment by kyeah** [comment]
yeah, go for it!

**Comment by saghm** [comment]
I'm thinking we should push a new version to crates?

**Comment by kyeah** [comment]
yup, looks like a good time for push out 0.1.5.

**Pull request merged**